### ISSUE TITLE
Centered window placement on high res displays

### ISSUE THREAD

**Issue by sargunv** [godotengine/godot#70752]
### Godot version

v3.5.1.stable.official [6fed1ffa3]

### System information

Windows 11 set to 150% scaling on a 5120x2160 display

### Issue description

By default, when I test a scene in the Godot editor, the test window appears somewhere near the bottom right of my display, partially off screen. I have to drag it back within view every time I test.

Upon investigation, I found the Editor Settings -> Run -> Window Placement -> Rect setting, which is set to Centered. Setting to Top Left works correctly, but Centered does not.

With some experimentation, I found the Project Settings -> Display -> Window -> DPI -> Allow hiDPI setting, when set to "On", causes the test window to correctly appear in the center of the screen, but the window is now too small (since my game isn't programmed to adapt to hiDPI yet).

If I disable scaling in Windows and restart the editor, the window centers properly (but of course everything is now too small).

So, it seems the editor's Centered setting seems to incorrectly calculate the center of the screen when scaling is enabled in Windows unless the "Allow hiDPI" setting is also enabled in the project.

I've worked around it for now by setting a "Custom Position" window placement instead, but I have to tweak this depending on the screen (laptop or monitor) and specific project, which isn't ideal.

### Steps to reproduce

![image](https://user-images.githubusercontent.com/1320357/210112143-718113cd-1e91-4814-ae8c-6016f977de26.png)

![image](https://user-images.githubusercontent.com/1320357/210112620-6347958f-3209-4c6b-ac83-91eddda345f8.png)

![image](https://user-images.githubusercontent.com/1320357/210112670-651a9b31-27f3-4619-8132-38d2e462e8fd.png)

### Minimal reproduction project

N/A

**Comment by KoBeWi** [comment]
Duplicate of #13017, but the issue was resolved by making "Allow hiDPI" enabled by default. Judging from your comments about small window, such solution is not ideal?

Does the issue happen in Godot 4?

**Comment by Calinou** [comment]
Low-DPI fallbacks are considered unsupported in modern operating systems. Feel free to try fixing this, but it may not be worth the effort (especially with multi-monitor setups).

**Comment by sargunv** [comment]
> Duplicate of https://github.com/godotengine/godot/issues/13017, but the issue was resolved by making "Allow hiDPI" enabled by default. Judging from your comments about small window, such solution is not ideal?

Yes, enabling "Allow hiDPI" means the project has to include the functionality to detect the native monitor size/resolution and dynamically size up the window relative to that, or the window is too small. My game is currently set to an exact test window of 768x672 (3x the pixel resolution of 256x224).

Ideally, the editor should be able to correctly center any project's test window, regardless of whether the project enabled "Allow hiDPI".

I've never worked on Godot's codebase, but I'd be happy to investigate/fix this myself with perhaps a pointer in the general area I should start looking at.

**Comment by sargunv** [comment]
>At the same time, I suggest making the default window width and height automatically scale if Allow HiDPI is enabled (regardless of Test Width/Test Height are used or not). This will be useful in 4.0 as well.

This is fair, but imo if the option is offered it should work correctly in both modes. Totally understand this not being worth the effort though. Still happy to look into it myself as possibly a way to dip my toes into Godot development.

**Comment by Calinou** [comment]
Running window position on startup is set here in 4.0 (does not affect exported projects): https://github.com/Calinou/godot/blob/25c1d23110d2a5ddeb0902cc044fc3dceee50c5f/editor/editor_run.cpp#L155-L183

And in 3.x: https://github.com/godotengine/godot/blob/3.x/editor/editor_run.cpp#L143-L171

___

Window size on startup is set here in 4.0 (both when running from the editor and in exported projects): https://github.com/godotengine/godot/blob/master/main/main.cpp#L1681-L1696

And in 3.x: https://github.com/godotengine/godot/blob/3.x/main/main.cpp#L1118-L1132

___

Note that there is no way to get the OS scale factor on Windows yet (4.0's `DisplayServer.screen_get_scale()` or 3.x's `OS.get_screen_scale()` is only effective on macOS). For now, you could assume that the editor's scale factor setting is configured to match the OS scale factor setting, but this isn't always true in practice: https://github.com/godotengine/godot/blob/3.x/editor/project_manager.cpp#L2406-L2436
